rpc/block.go

```diff
@@ -155,9 +155,10 @@ type BlockMetadata struct {
 
 // GetBlock returns information about a Tezos block
 // https://tezos.gitlab.io/mainnet/api/rpc.html#get-block-id
-func (c *Client) GetBlock(ctx context.Context, blockID tezos.BlockHash) (*Block, error) {
+func (c *Client) GetBlock(ctx context.Context, blockID interface{}) (*Block, error) {
```

Comment:
Using an `interface{}` here would break type safety which is not desirable.

Inspired by the changes you intended, we added two new methods `GetHeadBlock()` and `GetGenesisBlock()` which we consider the canonical way of extending TzGo in this case.